--- lib/requests/models.py	2018-01-03 14:10:27.000000000 -0800
+++ models.py	2018-01-03 14:09:01.000000000 -0800
@@ -742,8 +742,16 @@
             # Special case for urllib3.
             if hasattr(self.raw, 'stream'):
                 try:
-                    for chunk in self.raw.stream(chunk_size, decode_content=True):
-                        yield chunk
+                    if isinstance(self.raw._original_response._method, int):
+                        while True:
+                            chunk = self.raw.read(chunk_size, decode_content=True)
+                            if not chunk:
+                                break
+                            yield chunk
+                    else:
+                        for chunk in self.raw.stream(chunk_size, decode_content=True):
+                            yield chunk
+
                 except ProtocolError as e:
                     raise ChunkedEncodingError(e)
                 except DecodeError as e:
